flow-report/Sidebar.tsx

```diff
@@ -0,0 +1,155 @@
+      href={href}
+    >
+      <div className="SidebarFlowStep_icon">
+        <div
+          className="SidebarFlowStep_icon_line"
+          style={hideTopLine ? {background: 'transparent'} : undefined}
+        />
+        <FlowStepIcon mode={mode}/>
+        <div
+          className="SidebarFlowStep_icon_line"
+          style={hideBottomLine ? {background: 'transparent'} : undefined}
+        />
+      </div>
+      <div className={`SidebarFlowStep_label ${mode}`}>{label}</div>
+    </a>
+  );
+};
+
+export const SidebarFlow: FunctionComponent<{flowResult: LH.FlowResult}> = ({flowResult}) => {
+  let numNavigation = 1;
+  let numTimespan = 1;
+  let numSnapshot = 1;
+  const steps = flowResult.lhrs.map((lhr, index) => {
+    let name;
+    switch (lhr.gatherMode) {
+      case 'navigation':
+        name = `Navigation (${numNavigation++})`;
+        break;
+      case 'timespan':
+        name = `Timespan (${numTimespan++})`;
+        break;
+      case 'snapshot':
+        name = `Snapshot (${numSnapshot++})`;
+        break;
+    }
+    const url = new URL(location.href);
+    url.searchParams.set('step', String(index));
+    const currentStep = useCurrentStep();
+    return (
+      <SidebarFlowStep
+        key={lhr.fetchTime}
+        mode={lhr.gatherMode}
+        href={url.href}
+        label={name}
+        hideTopLine={index === 0}
+        hideBottomLine={index === flowResult.lhrs.length - 1}
+        isCurrent={index === currentStep}
+      />
+    );
+  });
+  return (
+    <>
+      {steps}
+    </>
+  );
+};
+
+export const SidebarRuntimeSettings: FunctionComponent<{settings: LH.Config.Settings}> =
+({settings}) => {
+  return (
+    <details className="SidebarRuntimeSettings">
+      <summary>
+        {
+          `${settings.screenEmulation.height}x${settings.screenEmulation.width}px | ` +
+          `${settings.formFactor}`
+        }
+      </summary>
+      <div>Emulated user agent: {settings.emulatedUserAgent}</div>
+      <div>Channel: {settings.channel}</div>
+    </details>
+  );
+};
+
+export const SidebarHeader: FunctionComponent<{title: string, date: string}> = ({title, date}) => {
+  const formatter = useMemo(() => {
+    const options:Intl.DateTimeFormatOptions = {
+      month: 'short', day: 'numeric', year: 'numeric',
+      hour: 'numeric', minute: 'numeric', timeZoneName: 'short',
+    };
+    return new Intl.DateTimeFormat('en-US', options);
+  }, []);
+  const dateString = useMemo(() => formatter.format(new Date(date)), [date]);
+  return (
+    <div className="SidebarHeader">
+      <div className="SidebarHeader_title">{title}</div>
+      <div className="SidebarHeader_date">{dateString}</div>
+    </div>
+  );
+};
+
+export const SidebarSectionTitle: FunctionComponent = ({children}) => {
+  return <div className="SidebarSectionTitle">{children}</div>;
+};
+
+export const Sidebar: FunctionComponent<{flowResult: LH.FlowResult}> = ({flowResult}) => {
```

Comment:
instead of exporting everything, can you export only what is meant to be used directly / what cannot be tested indirectly?